repos/keg-components/src/components/form/checkbox/checkbox.wrapper.js

```diff
@@ -86,22 +101,50 @@ const ChildrenComponent = ({ children, className }) => (
  * CheckboxWrapper
  * Wraps the Passed in Checkbox Element
  * @param {Object} props - see PropTypes below
- *
+ * @param {string?} props.className - css class string
+ * @param {boolean?} props.initChecked - the initial checked/unchecked value of Checkbox
+ * @param {boolean?} props.checked - alias for initChecked
+ * @param {Object | String | Array} props.children - react children
+ * @param {string} props.elType - type of checkbox
+ * @param {Component} props.Element - the which component used for the core checkbox ui. @see checkbox.js
+ * @param {Component?} props.CheckIcon - an optional component that overrides the icon used in the checkbox
+ * @param {boolean} props.disabled - disables checking and unchecking if true
+ * @param {boolean} props.disableCheck - disables checking if true
+ * @param {boolean} props.disableUncheck - disables unchecking if true
+ * @param {boolean} props.allowAdjacentPress - if true, allows pressing a side component to toggle the checkbox
+ * @param {boolean} props.isWeb - indicates current platform is web. Provide by checkbox.js
+ * @param {(Component | string)?} props.LeftComponent - an optional component or string to render left of the checkbox
+ * @param {boolean} props.close - if true, side components will be rendered close to the checkbox. If false, spaced apart.
+ * @param {Function?} props.onChange - callback fired when checkbox is toggled
+ * @param {Function?} props.onValueChange - callback fired when checkbox is toggled
+ * @param {Function?} props.setCheckedSetter - callback called once on initial render. Is passed the `setChecked` callback to provide consumer control of checked/unchecked state if needed.
+ * @param {(Component | string)?} props.RightComponent -  an optional component or string to render right of the checkbox 
+ * @param {Object} props.styles - styles that overwrite default checkbox styles
+ * @param {Component?} props.CheckboxComponent - an optional component that overrides the default checkbox element, passed the same props and styles
+ * @param {string} props.type - theme type
+ * @param {string} props.themePath - path to theme object
+ * @param {boolean} props.value - alias for initChecked
+ * @param {...rest} props.* - the remaining props are passed to Element
  */
 export const CheckboxWrapper = props => {
   const {
     className,
+    initChecked,
     checked,
     children,
     elType,
     Element,
+    CheckIcon,
     disabled,
+    disableCheck = false,
+    disableUncheck = false,
+    allowAdjacentPress = true,
     isWeb,
     LeftComponent,
     close,
     onChange,
     onValueChange,
-    ref,
+    setCheckedSetter,
```

Comment:
This prop name seems off. I get what it's doing but I think it could be better.
The set part does not make sense, cause really it's just calling a function. It's not setting anything with an `=`
Would be better to use a `ref` and set the ref which follow the rest of the react paradigm. For example =>

```js
// Consumer calls child checkbox
const ConsumerComp = props => {
  const toggleCheckedRef = userRef(null)
  <Checkbox toggleCheckedRef={ toggleCheckedRef } /> 
}

// Then in the wrapper set the ref in the useEffect hook
const CheckboxWrapper = props => {
  // ...other codes

  useEffect(() => {
    toggleCheckedRef && ( toggleCheckedRef.current = setChecked )
  }, [ toggleCheckedRef, setChecked ])
}

```

Or event, just set it inline with out the hook, since we have access to it => 
```js
const CheckboxWrapper = props => {
  // ...other codes
  toggleCheckedRef && ( toggleCheckedRef.current = setChecked )
}
```

Reply:
Yeah, I agree it was a bit confusing.

I've changed the checkbox and the group to work with refs.

I decided to use the `useImperativeHandle` hook, to define the api the consumer can use (in this case, `setChecked` or `isChecked`). I prefer that because we expose only what the consumer needs, it uses the standard ref argument of react, and I think it's  makes clear what is going on.

 Let me know what you think.